services/core-api/app/api/help/resources/help_resource.py

```diff
@@ -0,0 +1,138 @@
+from flask_restx import Resource, reqparse
+from flask import request
+from app.api.utils.access_decorators import requires_any_of, VIEW_ALL, MINESPACE_PROPONENT, EDIT_HELPDESK, is_minespace_user
+from app.extensions import api
+from werkzeug.exceptions import BadRequest
+from enum import Enum
+
+from app.api.utils.resources_mixins import UserMixin
+from app.api.help.models.help import Help
+from app.api.help.response_models import HELP_MODEL
+
+class SystemFlag(Enum):
+    core = "CORE"
+    ms = "MineSpace"
+
+    def __str__(self):
+        return self.value
+    
+class HelpListResource(Resource, UserMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument(
+        'system',
+        type=str,
+        help='The system (CORE or MineSpace) where the guide will be displayed',
+        location='json',
+        required=False,
+    )
+    
+
+    @api.doc(
+        description="List all help guides",
+    )
+    @api.marshal_with(HELP_MODEL, code=200, as_list=True, envelope="records")
+    @requires_any_of([VIEW_ALL, MINESPACE_PROPONENT])
+    def get(self):
+        data = self.parser.parse_args()
+        system = data.get("system", None) if not is_minespace_user() else str(SystemFlag.ms)
+
+        help_guides = Help.get_all(system)
+        return help_guides    
+
+class HelpResource(Resource, UserMixin):
+
+    parser = reqparse.RequestParser()
+    parser.add_argument(
+        'system',
+        type=str,
+        help='The system (CORE or MineSpace) where the guide will be displayed',
+        location='json',
+        required=False,
+    )
+    parser.add_argument(
+        'page_tab',
+        type=str,
+        help='The specific tab on the page that the help guide is for',
+        location='json',
+        required=False,
+    )
+    parser.add_argument(
+        'content',
+        type=str,
+        help='Help user guide content',
+        location='json',
+        required=False,
+    )
+    parser.add_argument(
+        'is_draft',
+        type=bool,
+        help='Is the guide in a draft state',
+        location='json',
+        required=False,
+    )
+    parser.add_argument(
+        'help_guid',
+        type=str,
+        help='The identifier of the help guide to modify',
+        location='json',
+        required=False,
+    )    
+
+    @requires_any_of([VIEW_ALL, MINESPACE_PROPONENT])
+    @api.marshal_with(HELP_MODEL, code=200, envelope='records', as_list=True)
+    def get(self, help_key):
+        system = request.args['system']
+        if is_minespace_user() and system != str(SystemFlag.ms):
+            raise BadRequest("Only MineSpace help guides may be requested.")
+        help_guides = Help.find_by_help_key(help_key, system)
+        return help_guides
+    
+    @requires_any_of([EDIT_HELPDESK])
+    @api.expect(parser)
+    @api.marshal_with(HELP_MODEL, code=201)
+    def post(self, help_key):
```

Comment:
Would be great if we could add some tests to these endpoints, and add the resource + expected auth to `tests/auth/test_expected_auth.py`